Adding details for QDR password auth (#502)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/proc_retrieving-the-qdr-password.adoc

[id="retrieving-the-qdr-password_{context}"]
= Retrieving the {MessageBus} password

[role="_abstract"]
When you configure the {OpenStack} ({OpenStackShort}) overcloud for {Project} ({ProjectShort}), you must provide the {MessageBus} password in the {ProjectShort} connection file.

> Basic authentication on the {MessageBus} connection can be disabled by setting `transports.qdr.auth` parameter of the STF spec to `none`. Versions of {ProjectShort} before 1.5.3 did not support this parameter and always behave as if it is set to `none`. This behaviour is preserved after an upgrade to {ProjectShort} 1.5.3 until a user explicitly changes the parameter to `basic`. New installs of {ProjectShort} 1.5.3 or later use `basic` authentication by default.

Reviewer: ```suggestion
Basic authentication on the {MessageBus} connection can be disabled by setting `transports.qdr.auth` parameter of the ServiceTelemetry spec to `none`. Versions of {ProjectShort} before 1.5.3 did not support this parameter and will always behave as if set to `none`. This behavior is preserved after an upgrade to {ProjectShort} 1.5.3 until a user explicitly changes the parameter to `basic`. New installs of {ProjectShort} 1.5.3 or later use `basic` authentication by default.
```

Reviewer: 🇨🇦 🍁

> Basic authentication on the {MessageBus} connection can be disabled by setting `transports.qdr.auth` parameter of the ServiceTelemetry spec to `none`. Versions of {ProjectShort} before 1.5.3 did not support this parameter and will always behave as if set to `none`. This behavior is preserved after an upgrade to {ProjectShort} 1.5.3 until a user explicitly changes the parameter to `basic`. New installs of {ProjectShort} 1.5.3 or later use `basic` authentication by default.

Reviewer 2: ```suggestion
You can disable basic authentication on the {MessageBus} connection by setting the value of the `transports.qdr.auth` parameter of the ServiceTelemetry spec to `none`. The `transports.qdr.auth` parameter is absent in versions of {ProjectShort} before 1.5.3, so the default behavior is that basic authentication is disabled. In a new install of {ProjectShort} 1.5.3 or later, the default value of  `transports.qdr.auth` is `basic`, but if you upgraded to {ProjectShort} 1.5.3, the default value of `transports.qdr.auth` is `none`.
```

Author: This is actually much better, thank you.

[id="retrieving-the-qdr-password_{context}"]
= Retrieving the {MessageBus} password

[role="_abstract"]
When you configure the {OpenStack} ({OpenStackShort}) overcloud for {Project} ({ProjectShort}), you must provide the {MessageBus} password in the {ProjectShort} connection file.

.Procedure

. Log in to your {OpenShift} environment where {ProjectShort} is hosted.

. Change to the `service-telemetry` project:
+
[source,bash]
----
$ oc project service-telemetry
----

. Retrieve the {MessageBus} password:
+
[source,bash,options="nowrap",subs="verbatim"]
----
$ oc get secret default-interconnect-users -o json | jq -r .data.guest | base64 -d
----

[NOTE]

> Basic authentication on the {MessageBus} connection can be disabled by setting `transports.qdr.auth` parameter of the STF spec to `none`. Versions of {ProjectShort} before 1.5.3 did not support this parameter and always behave as if it is set to `none`. This behaviour is preserved after an upgrade to {ProjectShort} 1.5.3 until a user explicitly changes the parameter to `basic`. New installs of {ProjectShort} 1.5.3 or later use `basic` authentication by default.

Reviewer: I think I'd move this up to the description part of the procedure rather than at the end.
